src/core/Akka.Tests/Serialization/SerializeAllMessagesSpec.cs

```diff
@@ -0,0 +1,76 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SerializeAllMessagesSpec.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2023 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2023 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Event;
+using Akka.TestKit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Tests.Serialization;
+
+public class SerializeAllMessagesSpec : AkkaSpec
+{
+    public SerializeAllMessagesSpec(ITestOutputHelper output) : base("akka.actor.serialize-messages = on", output)
+    {
+    }
+    
+    private class MyMessage : INoSerializationVerificationNeeded
+    {
+        public MyMessage(Action myAction)
+        {
+            MyAction = myAction;
+        }
+
+        // add an unserializable member, such as a delegate
+        public Action MyAction { get; }
+    }
+
+    private class MyWrappedMessage : IWrappedMessage
+    {
+        public MyWrappedMessage(object message)
+        {
+            Message = message;
+        }
+
+        public object Message { get; }
+    }
+    
+    // create an actor that will process a MyWrappedMessage and invoke the delegate
+    private class MyActor : ReceiveActor
+    {
+        public MyActor()
+        {
+            Receive<MyWrappedMessage>(msg =>
+            {
+                var myMessage = (MyMessage) msg.Message;
+                myMessage.MyAction();
+            });
+        }
+    }
+    
+    [Fact]
+    public async Task Should_not_serialize_WrappedMessage_with_INoSerializationVerificationNeeded()
+    {
+        // Arrange
+        var myProbe = CreateTestProbe();
+        var action = () => { myProbe.Tell("worked"); };
+        var message = new MyMessage(action);
+        var wrappedMessage = new MyWrappedMessage(message);
+        
+        var myActor = Sys.ActorOf(Props.Create(() => new MyActor()), "wrapped-message-actor");
+
+        await EventFilter.Error().ExpectAsync(0, async () =>
+        {
+            // Act
+            myActor.Tell(wrappedMessage);
+            await myProbe.ExpectMsgAsync("worked");
```

Comment:
Both the `EventFilter` will fail and the `ExpectMsgAsync` will fail if the `ActorCell` of the recipient attempts to serialize this message - which is exactly what happened before I applied my fix.